### PR TITLE
remove --force-reinstall from Dockerfile to ensure correct pytorch version

### LINKED ISSUE #444: Docker Hub images appear to install Torch with CUDA 11.7 instead of CUDA 11.8
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Using `docker pull winglian/axolotl:main-py3.10-cu118-2.0.1` or `docker pull winglian/axolotl:main-py3.9-cu118-2.0.1`, the Torch version installed should use CUDA 11.8, in accordance with the BNB install which is compiled for 11.8.

### Current behaviour

I'm using the Docker images with boxes from vast.ai. I've tested both 3.9 and 3.10, and both appear to be installed with Torch 11.7 by default:

```
root@C.6794639:~$ python3 -c 'import torch; print(torch.version.cuda)'
11.7

root@C.6794639:~$ python3 -m pip install --no-cache-dir -U torch==2.0.1+cu118 --extra-index-url https://download.pytorch.org/whl/cu118
# ...
Installing collected packages: torch
  Attempting uninstall: torch
    Found existing installation: torch 2.0.1
    Uninstalling torch-2.0.1:
      Successfully uninstalled torch-2.0.1
Successfully installed torch-2.0.1+cu118

root@C.6794639:~$ python3 -c 'import torch; print(torch.version.cuda)'
11.8
```

This means that Axolotl doesn't actually work for QLoRA training on these images out of the box, as BNB is compiled for CUDA 11.8 and gets confused by Torch being compiled for CUDA 11.7.

### Steps to reproduce

Pull down the image from the Docker Hub, then check the installed version of Torch with `python3 -c 'import torch; print(torch.version.cuda)'`.

### Possible solution

Unfortunately, Docker Hub doesn't give me much visibility into what's going on with the individual build steps, but my guess is that string interpolation is failing and Torch is being installed with 11.7. Installing Torch with 11.8 manually using the command above (noting that I took that from the base Dockerfile and manually interpolated the arguments) appears to work fine.

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.9 / 3.10

### axolotl branch-commit

main/9d629d8bff1c69a8d7753b18ab6029a7aaea2090

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

### PR DESCRIPTION
**Pull request by tmm1**
fixes #444

### CODE REVIEW

**Review by winglian — APPROVED** [review]
Once the old gptq is stripped out, this isn't needed and can be installed normally
